test/libyul/ControlFlowGraphTest.cpp

```diff
@@ -0,0 +1,270 @@
+				m_stream << stackSlotToString(_conditionalJump.condition);
+				m_stream << "| { <0> Zero | <1> NonZero }}\" shape=Mrecord];\n";
+				m_stream << "Block" << getBlockId(_block);
+				m_stream << "Exit:0 -> Block" << getBlockId(*_conditionalJump.zero) << ";\n";
+				m_stream << "Block" << getBlockId(_block);
+				m_stream << "Exit:1 -> Block" << getBlockId(*_conditionalJump.nonZero) << ";\n";
+			},
+			[&](CFG::BasicBlock::FunctionReturn const& _return)
+			{
+				m_stream << "Block" << getBlockId(_block) << "Exit [label=\"FunctionReturn[" << _return.info->function.name.str() << "]\"];\n";
+				m_stream << "Block" << getBlockId(_block) << " -> Block" << getBlockId(_block) << "Exit;\n";
+			},
+			[&](CFG::BasicBlock::Terminated const&)
+			{
+				m_stream << "Block" << getBlockId(_block) << "Exit [label=\"Terminated\"];\n";
+				m_stream << "Block" << getBlockId(_block) << " -> Block" << getBlockId(_block) << "Exit;\n";
+			}
+		}, _block.exit);
+		m_stream << "\n";
+	}
+	size_t getBlockId(CFG::BasicBlock const& _block)
+	{
+		if (size_t* id = util::valueOrNullptr(m_blockIds, &_block))
+			return *id;
+		size_t id = m_blockIds[&_block] = m_blockCount++;
+		m_blocksToPrint.emplace_back(&_block);
+		return id;
+	}
+	std::ostream& m_stream;
+	std::map<CFG::BasicBlock const*, size_t> m_blockIds;
+	size_t m_blockCount = 0;
+	std::list<CFG::BasicBlock const*> m_blocksToPrint;
+};
+
+TestCase::TestResult ControlFlowGraphTest::run(ostream& _stream, string const& _linePrefix, bool const _formatted)
+{
+	ErrorList errors;
+	auto [object, analysisInfo] = parse(m_source, *m_dialect, errors);
+	if (!object || !analysisInfo || !Error::containsOnlyWarnings(errors))
+	{
+		AnsiColorized(_stream, _formatted, {formatting::BOLD, formatting::RED}) << _linePrefix << "Error parsing source." << endl;
+		printErrors(errors);
+		return TestResult::FatalError;
+	}
+
+	std::ostringstream output;
+
+	std::unique_ptr<CFG> cfg = ControlFlowGraphBuilder::build(*analysisInfo, *m_dialect, *object->code);
+
+	output << "digraph CFG {\nnodesep=0.7;\nnode[shape=box];\n\n";
+	ControlFlowGraphPrinter printer{output};
+	printer(*cfg->entry);
+	for (auto function: cfg->functions)
+		printer(cfg->functionInfo.at(function));
+	output << "}\n";
+
+	m_obtainedResult = output.str();
+
+	auto result = checkResult(_stream, _linePrefix, _formatted);
+
+#ifdef ISOLTEST
+	char* graphDisplayer = nullptr;
+	if (result == TestResult::Failure)
+		graphDisplayer = getenv("ISOLTEST_DISPLAY_GRAPHS_FAILURE");
+	else if (result == TestResult::Success)
+		graphDisplayer = getenv("ISOLTEST_DISPLAY_GRAPHS_SUCCESS");
+
+	if (graphDisplayer)
+	{
+		if (result == TestResult::Success)
+			std::cout << std::endl << m_source << std::endl;
+		boost::process::opstream pipe;
+		boost::process::child child(graphDisplayer, boost::process::std_in < pipe);
+
+		pipe << output.str();
+		pipe.flush();
+		pipe.pipe().close();
+		if (result == TestResult::Success)
+			child.wait();
+		else
+			child.detach();
```

Comment:
If you detach on failure, won't this just result in isoltest leaving spawned child processes running in the background?

Reply:
Yes. In the failure case, ``isoltest`` will display the test case and a ``diff`` and stop on its own, but only after you return from here, so this is intentional.
In the success case it won't stop on its own and not output anything, so I output the source manually here and have the child process block.